tests/rohf-orb-rot/input.dat

```diff
@@ -0,0 +1,129 @@
+#!   Test if DGAS's orbital rotation code matches expected values.
+#!   The first two calculations compute the X (symm A'') and A (symm A') states of HO2 in Cs
+#! by restricting the occupations of the two states.
+#!   The second two calculations compute the X and A states of HO2 in C1. Computation
+#! of the A state requires a rotation of the HOMO and SOMO by 90 degrees
+#! (i.e., by swapping these two orbitals). The orb_rotate() function accomplishes this.
+
+# Restrict occupations to compute CCSD/6-31G energy of X state (symm A'') of HO2 in Cs.
+set { 
+    basis 6-31G
+    reference rohf
+    maxiter 500
+    e_convergence 10
+    d_convergence 10
+    r_convergence 10
+    memory 60000
+    freeze_core True
+    scf_type pk
+    docc [7, 1]
+    socc [0, 1]
+}
+
+molecule ho2 {
+    0 2
+    O -0.6937342972  0.0081785728  0.0000000000
+    O  0.6354757377 -0.0626608366  0.0000000000
+    H  0.9246056151  0.8646730647  0.0000000000
+}
+
+scf_e, scf_wfn = energy('scf', return_wfn=True)
+ccsd_e, ccsd_wfn = energy('ccsd', return_wfn=True)
+
+compare_values(-150.108136116503147, scf_e, 6, 'X SCF energy') #TEST
+compare_values(-150.347533153677489, ccsd_e, 6, 'X CCSD energy') #TEST
+
+clean()
+
+# Restrict occupations to compute CCSD/6-31G energy of A state (symm A') of HO2 in Cs.
+set {
+    docc [6, 2]
+    socc [1, 0]
+}
+
+scf_e, scf_wfn = energy('scf', return_wfn=True)
+ccsd_e, ccsd_wfn = energy('ccsd', return_wfn=True)
+
+compare_values(-150.087298715913619, scf_e, 6, 'A SCF energy') #TEST
+compare_values(-150.312480530946573, ccsd_e, 6, 'A CCSD energy') #TEST
+
+clean()
+
+# Compute CCSD/6-31G energy of X state of HO2 in C1.
+set {
+    docc [8]
+    socc [1]
+}
+
+molecule ho2 {
+    0 2
+    O -0.6937342972  0.0081785728  0.0000000000
+    O  0.6354757377 -0.0626608366  0.0000000000
+    H  0.9246056151  0.8646730647  0.0000000000
+    symmetry c1
+}
+
+scf_e, scf_wfn = energy('scf', return_wfn=True)
+ccsd_e, ccsd_wfn = energy('ccsd', return_wfn=True)
+
+compare_values(-150.108136116503147, scf_e, 6, 'X SCF energy') #TEST
+compare_values(-150.347533153677489, ccsd_e, 6, 'X CCSD energy') #TEST
+
+# Rotate HOMO and SOMO by 90 degrees to obtain guess for A state of HO2 in C1.
+
+# Quick orbital rotation code.
+def orb_rotate(C, h, i, j, theta):
+
+   if h > C.nirrep():
+       raise p4util.PsiException("ROTATE: Expression %s irrep number is larger than the number of irreps" %
+                               (str(h)))
+
+   if max(i, j) > C.coldim()[h]:
+       raise p4util.PsiException("ROTATE: Expression %s orbital number exceeds number of orbitals in irrep" %
+                               (str(max(i, j))))
+
+   theta = np.deg2rad(theta)
+
+   x = C.nph[h][:, i].copy()
+   y = C.nph[h][:, j].copy()
+
+   xp = np.cos(theta) * x - np.sin(theta) * y
+   yp = np.sin(theta) * x + np.cos(theta) * y
+
+   C.nph[h][:, i] = xp
+   C.nph[h][:, j] = yp
+
+# Rotate columns here.
+orb_rotate(scf_wfn.Ca(), 0, 7, 8, 90.0)
+
```

Comment:
I think this can be switched to use the new `void rotate_columns(int h, int i, int j, double theta);` method on `psi4.Matrix` that @dgasmith mentioned.